Python/pywarpx/picmi.py

```diff
@@ -121,25 +126,45 @@ def initialize_inputs(self, layout,
         if initialize_self_fields is None:
             initialize_self_fields = False
 
-        self.species = pywarpx.Bucket.Bucket(self.name,
-                                             mass = self.mass,
-                                             charge = self.charge,
-                                             injection_style = None,
-                                             initialize_self_fields = int(initialize_self_fields),
-                                             boost_adjust_transverse_positions = self.boost_adjust_transverse_positions,
-                                             self_fields_required_precision = self.self_fields_required_precision,
-                                             self_fields_absolute_tolerance = self.self_fields_absolute_tolerance,
-                                             self_fields_max_iters = self.self_fields_max_iters,
-                                             self_fields_verbosity = self.self_fields_verbosity,
-                                             save_particles_at_xlo = self.save_particles_at_xlo,
-                                             save_particles_at_xhi = self.save_particles_at_xhi,
-                                             save_particles_at_ylo = self.save_particles_at_ylo,
-                                             save_particles_at_yhi = self.save_particles_at_yhi,
-                                             save_particles_at_zlo = self.save_particles_at_zlo,
-                                             save_particles_at_zhi = self.save_particles_at_zhi,
-                                             save_particles_at_eb = self.save_particles_at_eb,
-                                             save_previous_position = self.save_previous_position,
-                                             do_not_deposit = self.do_not_deposit)
+        if self.species_type == 'photon':
```

Comment:
The same question as above - is there a reason that photon species are handled separately?

Reply:
Having the separate `photon_species` list is good, but this separate setup of self.species is not needed. The `self.species.species_type` attribute isn't used anywhere.